socorro/processor/processor_app.py

```diff
@@ -242,25 +242,30 @@ def _transform(self, crash_id):
             exc_type, exc_value, exc_tb = sys.exc_info()
 
             # PolyStorage can throw a PolyStorageException which is a sequence
-            # of exc_info items.
+            # of exc_info items, so we need to capture each one
             if isinstance(exc_value, collections.Sequence):
                 exc_info = exc_value
             else:
                 exc_info = [(exc_type, exc_value, exc_tb)]
 
             for exc_info_item in exc_info:
                 self._capture_error(crash_id, exc_info_item)
+                self.config.logger.warning(
+                    'error in processing or saving crash %s',
+                    crash_id
+                )
 
             # Re-raise the original exception with the correct traceback
-            reraise(exc_type, exc_value, exc_tb)
+            six.reraise(exc_type, exc_value, exc_tb)
+
         finally:
             # Clean up any dump files saved to the file system
             for a_dump_pathname in dumps.values():
-                try:
-                    if 'TEMPORARY' in a_dump_pathname:
+                if 'TEMPORARY' in a_dump_pathname:
+                    try:
                         os.unlink(a_dump_pathname)
-                except OSError as x:
-                    self.config.logger.info('deletion of dump failed: %s', x)
+                    except OSError as x:
+                        self.config.logger.info('deletion of dump failed: %s', x)
```

Comment:
I inverted the if block because if `TEMPORARY` isn't in the name, then there's no sense in doing the work to set up exception handling.